testsuite/integration/manualmode/src/test/java/org/wildfly/test/manual/elytron/oidc/KeycloakConfiguration.java

```diff
@@ -0,0 +1,166 @@
+     * <li>user named bob and password bob123+ with "user" role</li>
+     * </ul>
+     * </ul>
+     */
+    public static RealmRepresentation getRealmRepresentation(final String realmName, String clientSecret,
+                                                             String clientHostName, int clientPort, Map<String, ClientAppType> clientApps) {
+        return createRealm(realmName, clientSecret, clientHostName, clientPort, clientApps);
+    }
+
+    public static String getAdminAccessToken(String authServerUrl) {
+        RequestSpecification requestSpecification = RestAssured
+                .given()
+                .param("grant_type", "password")
+                .param("username", KeycloakContainer.ADMIN_USER)
+                .param("password", KeycloakContainer.ADMIN_PASSWORD)
+                .param("client_id", "admin-cli");
+
+        Response response = requestSpecification.when().post(authServerUrl + "/realms/master/protocol/openid-connect/token");
+
+        final long deadline = System.currentTimeMillis() + 180000;
+        while (response.getStatusCode() != 200) {
+            // the Keycloak admin user isn't available yet, keep trying until it is to ensure we can obtain the token
+            // needed to set up the realms for the test
+            response = requestSpecification.when().post(authServerUrl + "/realms/master/protocol/openid-connect/token");
+            if (System.currentTimeMillis() > deadline) {
+                return null;
+            }
+        }
+
+        return response.as(AccessTokenResponse.class).getToken();
+    }
+
+    private static RealmRepresentation createRealm(String name, String clientSecret,
+                                                   String clientHostName, int clientPort, Map<String, ClientAppType> clientApps) {
+        RealmRepresentation realm = new RealmRepresentation();
+
+        realm.setRealm(name);
+        realm.setEnabled(true);
+        realm.setUsers(new ArrayList<>());
+        realm.setClients(new ArrayList<>());
+        realm.setAccessTokenLifespan(3);
+        realm.setSsoSessionMaxLifespan(3);
+
+        RolesRepresentation roles = new RolesRepresentation();
+        List<RoleRepresentation> realmRoles = new ArrayList<>();
+
+        roles.setRealm(realmRoles);
+        realm.setRoles(roles);
+
+        realm.getRoles().getRealm().add(new RoleRepresentation(USER_ROLE, null, false));
+        realm.getRoles().getRealm().add(new RoleRepresentation(JBOSS_ADMIN_ROLE, null, false));
+
+        for (Map.Entry<String, ClientAppType> entry : clientApps.entrySet()) {
+            String clientApp = entry.getKey();
+            ClientRepresentation clientRepresentation = createWebAppClient(clientApp, clientSecret, clientHostName, clientPort, clientApp, false);
+//            clientRepresentation.setDefaultClientScopes(List.of("openid"));
+//            clientRepresentation.setOptionalClientScopes(Arrays.asList("profile", "email", "phone"));
```

Comment:
Adding this would be nice so that default scopes don't get added to the token automatically and we can check that these are configured only when we ask for it. But this makes the tests fail with a 403 error when simulating a buttonclick. 
Elytron testsuite already tests with this, so, could this be something we don't do on the wildfly testsuite.

Reply:
@PrarthonaPaul I think if we could configure this just for the client needed for the scope related tests, then the other tests that rely on some of these being present shouldn't be affected.

Reply:
testsuite/integration/manualmode only has scope-related tests. 
So, I think all clients that are created in this function are scope related. And those were the ones that were failing with the 403 on this line: 
https://github.com/wildfly/wildfly/blob/1353b48d82bb25fec85e0bc1f6e4f9ef7452975e/testsuite/integration/manualmode/src/test/java/org/wildfly/test/manual/elytron/oidc/OidcBaseTest.java#L175
I played around with the config and realised those lines were causing that failure.

Reply:
Thanks for the update, @PrarthonaPaul! Would you be able to create a WFLY issue for updating the default scopes that are used for these test cases and link that issue to WFLY-16532?

I think it should be possible to configure this (since we were able to do something similar from the corresponding tests in the Elytron testsuite).

Maybe something went wrong with the client creation or with adding the users to the realm or something like that. Debugging should help us see what's happening.

This doesn't need to block getting this merged though since we've already verified this from the Elytron testsuite so let's get the follow-up issue created to track this.

Reply:
https://issues.redhat.com/browse/WFLY-19176